src/test/ui/borrowck/issue-83760.stderr

```diff
@@ -0,0 +1,30 @@
+error[E0382]: use of moved value
+  --> $DIR/issue-83760.rs:5:20
+   |
+LL |     while let Some(foo) = val {
+   |                    ^^^ value moved here, in previous iteration of loop
+   |
+   = help: make sure that it is reinitialized on all paths
```

Comment:
I'm not entirely happy with the `help` wording because it feels like it might lead to confusion. It's likely better than we have already, but I'd feel better with a more newcomer friendly wording and potentially some spans pointing at the move locations when stating this.

Reply:
How about something like:
```
46 |     while let Some(_foo) = val { //~ ERROR use of moved value
   |                    ^^^^
   |                    |
   |                    the value might not be reinitialized at this point
   |                    value moved here, in previous iteration of loop
```
We could also print the spans of the reinits that might get skipped, though I think this would probably clutter the output.

Reply:
> We could also print the spans of the reinits that might get skipped, though I think this would probably clutter the output.

Could we see what that would look like in the tests we have available? Something we do in other cases is provide a full print out if few things are involved, and hide almost everything when they become too verbose. We could do something like that, but I'd like to see what it'd look like first.